bundles/org.openhab.binding.modbus.e3dc/src/main/java/org/openhab/binding/modbus/e3dc/internal/dto/DataConverter.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.modbus.e3dc.internal.dto;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.BitSet;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * The {@link DataConverter} Helper class to convert bytes from modbus into desired data format
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+public class DataConverter {
+    private static final long MAX_INT32 = (long) Math.pow(2, Integer.SIZE);
+
+    /**
+     * Get unit16 value from 2 bytes
+     *
+     * @param wrap
+     * @return
+     */
+    public static int getUInt16Value(ByteBuffer wrap) {
+        return ((wrap.get() & 0xff) << 8) | wrap.get() & 0xff;
+    }
+
+    public static long getLongValue(ByteBuffer wrap) {
+        long value = 0;
+        for (int i = 0; i < 4; i++) {
+            value = (value << 8) + (wrap.get() & 0xff);
+        }
+        return value;
```

Comment:
```suggestion
        return wrap.getLong();
```

Reply:
Hm, getLong reads 8 bytes which is too long. I applied your idea before: getInt and convert to unsigned long =>         
return Integer.toUnsignedLong(wrap.getInt());
This works fine in my unit tests